Example/ViewController.swift

```diff
@@ -48,7 +54,237 @@ class ViewController: UIViewController {
+            if tailPosition == .left && unprojectedCoordinate.x > mapView.bounds.width * 0.75 {
+                tailPosition = .right
+            } else if tailPosition == .right && unprojectedCoordinate.x < mapView.bounds.width * 0.25 {
+                tailPosition = .left
+            }
+
+            let imageName = tailPosition == .left ? "RouteInfoAnnotationLeftHanded" : "RouteInfoAnnotationRightHanded"
+            point.attributes = ["selected": false, "tailPosition": tailPosition.rawValue, "text": labelText, "imageName": imageName, "sortOrder": -index]
+
+            features.append(point)
         }
+
+        // add the annotation for the selected annotation last so it ends up ordered on top of the others
+        let labelText = annotationLabelForRoute(selectedRoute, tolls: routesContainTolls)
+
+        let point = MGLPointFeature()
+        point.coordinate = selectedRouteCoordinate
+        point.attributes = ["selected": true, "tailPosition": selectedRouteTailPosition.rawValue, "text": labelText, "imageName": selectedRouteTailPosition == .left ? "RouteInfoAnnotationLeftHanded-Selected" : "RouteInfoAnnotationRightHanded-Selected"]
+        features.append(point)
+
+        // add the features to the style
+        addRouteAnnotationSymbolLayer(features: features, style: style)
+    }
+
+    private let annotationLayerIdentifier = "RouteETAAnnotations"
+
+    private func addRouteAnnotationSymbolLayer(features: [MGLPointFeature], style: MGLStyle) {
+        let dataSource: MGLShapeSource
+        if let source = style.source(withIdentifier: annotationLayerIdentifier + "-source") as? MGLShapeSource {
+            dataSource = source
+        } else {
+            dataSource = MGLShapeSource(identifier: annotationLayerIdentifier + "-source", features: features, options: nil)
+            style.addSource(dataSource)
+        }
+
+        let shapeLayer: MGLSymbolStyleLayer
+
+        if let layer = style.layer(withIdentifier: annotationLayerIdentifier + "-shape") as? MGLSymbolStyleLayer {
+            shapeLayer = layer
+        } else {
+            shapeLayer = MGLSymbolStyleLayer(identifier: annotationLayerIdentifier + "-shape", source: dataSource)
+        }
+
+        shapeLayer.text = NSExpression(forKeyPath: "text")
+        let fontSizeByZoomLevel = [
+            13: NSExpression(forConstantValue: 16),
+            15.5: NSExpression(forConstantValue: 20)
+        ]
+        shapeLayer.textFontSize = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", fontSizeByZoomLevel)
+
+        shapeLayer.textColor = NSExpression(forConditional: NSPredicate(format: "selected == true"),
+                                            trueExpression: NSExpression(forConstantValue: UIColor.white),
+                     falseExpression: NSExpression(forConstantValue: UIColor.black))
+
+        shapeLayer.textFontNames = NSExpression(forConstantValue: ["DIN Pro Medium"])
```

Comment:
There are some good discussions to have on exactly what visual treatment we want. I did make the text size vary by zoom as I saw that being done in some other nav apps.

I'd prefer to investigate the designs in a separate ticket later though since it's bound to take a bit longer to come to agreement on the details.